Mathlib/Util/Superscript.lean

```diff
@@ -225,6 +225,61 @@ def scriptParser.formatter (name : String) (m : Mapping) (k : SyntaxNodeKind) (p
   | .ok newStack =>
     set { st with stack := stack ++ newStack }
 
+open PrettyPrinter.Delaborator
+
+/-- Returns the user-facing name of any constant or free variable. -/
+private def name : Expr → MetaM (Option Name)
+  | Expr.const name _ => pure (privateToUserName name)
+  | Expr.fvar name => Functor.map (pure ∘ privateToUserName) name.getUserName
+  | _ => pure none
+
+/-- Returns `true` if every character in `s` can be subscripted. -/
+private def isSubscriptable (s : Name) : Bool :=
+  s.toString.toList.all Mapping.subscript.toSpecial.contains
+
+/-- Returns `true` if every character in `s` can be superscripted. -/
+private def isSuperscriptable (s : Name) : Bool :=
+  s.toString.toList.all Mapping.superscript.toSpecial.contains
+
+/-- Applies the predicate `f` to every explicit argument of `e`. -/
+private def check_args (e : Expr) (f : Expr → DelabM Unit) :
+    DelabM Unit := do
+  let args := e.getAppArgs
+  let kinds ← getParamKinds e.getAppFn args
+  -- The function may be partially-applied. We only need to check the args we have.
+  guard <| args.size <= kinds.size
+  args.zipIdx.zip kinds |>.filter
+    (fun (_, kind) ↦ kind.isRegularExplicit) |>.forM
+    fun ((x, i), _) ↦ SubExpr.withNaryArg i <| f x
```

Comment:
Using `withNaryArg` is not safe here unless `e` is always `getExpr`; so you should drop the `e` argument from this function.